File: 1950-sign-of-the-product-of-an-array/1950-sign-of-the-product-of-an-array.cpp
class Solution {
public:
    
int arraySign(vector<int> &nums)
{
    int product = 1;
    for (int i = 0; i < nums.size(); i++)
    {
        if (nums.at(i) == 0)
    {
        return 0;
    }
    else if (product*nums.at(i) > 0)
    {
        product = 1;
    }
    else
    {
        product =-1;
    }
    }
    return product;
}

};

File: 1950-sign-of-the-product-of-an-array/1950-sign-of-the-product-of-an-array.java
class Solution {
    public int arraySign(int[] nums) {
    short countOfNeg = 0;
    int len = nums.length;
    for (int i = 0; i<len; i++){
    if (nums[i] == 0) return 0;
    if (nums[i] < 0) countOfNeg++;
    }
    return ((countOfNeg%2==0) ? 1: -1);
    }
}